### ISSUE TITLE
Upload error for 渋谷La.mama

### ISSUE THREAD

**Issue by nicolas-raoul** [commons-app/apps-android-commons#3651]
Set FakeGPS (or equivalent) to be near this music venue: https://goo.gl/maps/go8ncPKxnDQgWTAD8

1. Compile the app using the "structured-data" branch (not sure if needed)
2. Start the app
3. Go to "Nearby"
4. Select the item `渋谷La.mama` https://www.wikidata.org/wiki/Q64778538
5. Try to upload this picture for it: ![20200411_171501](https://user-images.githubusercontent.com/99590/79219678-aa3f8700-7e8d-11ea-9c72-6e145e2bfc2f.jpg)
6. Upload error:

![Screenshot_20200411-171724_Commons](https://user-images.githubusercontent.com/99590/79219087-90ea0b00-7e8c-11ea-9567-1a4e4b7bbee9.jpg)

Notice also the truncated name `渋谷La`. I have not experienced this with any other item/picture, there must be something special with this one. Sorry I don't have a logcat. I have not tried on master either.

**Comment by macgills** [comment]
```
            return uploadInterface.uploadFileToStash(fileNameRequestBody, tokenRequestBody, filePart)
                    .map(stashUploadResponse -> stashUploadResponse.getUpload());
```
`getUpload` is null I would say this has nothing to do with structured data though I did test on it.
Getting the error response from the API now

**Comment by macgills** [comment]
Right now I am guessing we should replace `.` with `_`

**Comment by macgills** [comment]
B-I-N-G-O

```
D/OkHttp: {"errors":[{"code":"verification-error","text":"File extension \".mama\" does not match the detected MIME type of the file (image/jpeg).","data":{"details":["filetype-mime-mismatch","mama","image/jpeg"]},"module":"upload"}],"docref":"See https://commons.wikimedia.org/w/api.php for API usage. Subscribe to the mediawiki-api-announce mailing list at &lt;https://lists.wikimedia.org/mailman/listinfo/mediawiki-api-announce&gt; for notice of API deprecations and breaking changes.","servedby":"mw1281"}
```

**Comment by macgills** [comment]
And I'd wager for the second error 
```
   @NonNull public String getDisplayTitle() {
        return filename != null ? getPageTitle().getDisplayTextWithoutNamespace().replaceFirst("[.][^.]+$", "") : "";
    }
```
this is the culprit

**Comment by macgills** [comment]
This is not related to structuredData so I will remove my assignment

**Comment by ashishkumar468** [comment]
Tagging @misaochan @neslihanturan @maskaravivek  for suggestions

**Comment by misaochan** [comment]
Hmmm, I've never encountered this problem, it seems to be specifically for that particular Wikidata item?

**Comment by nicolas-raoul** [comment]
For all such Wikidata items I guess. Maybe use that item or rename the sandbox item momentarily, and before upload modify title (but not the label(s)) to prevent the problem seen at https://github.com/commons-app/apps-android-commons/issues/3651#issuecomment-613396737?

**Comment by WoodyMats** [comment]
Hello! I'm new here and i would like to try solve this bug to contribute to the project! Thank you. :)

**Comment by shabar-shab** [comment]
@nicolas-raoul can I work on this?

**Comment by nicolas-raoul** [comment]
@shabar-shab Yes, thanks!
Please let us know about your progress every few days :-)

**Comment by i-am-atif** [comment]
@nicolas-raoul can I work on this?

**Comment by shabar-shab** [comment]
@i-am-atif  I was not able to reproduce this issue. did you reproduce this issue.

**Comment by i-am-atif** [comment]
@shabar-shab I just tried to reproduce this issue, but failed to do so

**Comment by i-am-atif** [comment]
I was able to upload the image without any error.

**Comment by nicolas-raoul** [comment]
@i-am-atif Did you follow all the steps exactly, especially step 4?
Please post the URL of your uploaded image here, thanks!

**Comment by sotiriszogos** [comment]
Is this issue still open?

**Comment by nicolas-raoul** [comment]
@sotiriszogos Yes! The first step would be to try and reproduce it, possibly by using a [sandbox item](https://www.wikidata.org/wiki/Q4115189).

**Comment by sotiriszogos** [comment]
@nicolas-raoul Thank you, I will take a look by the end of the week to reproduce it and work on it if noone is currently working on this issue.

**Comment by nicolas-raoul** [comment]
@shabar-shab Are you still working on this? Please let us know about anything you have found out :-)

**Comment by jmora13** [comment]
Could I work on this? Is it still available?

**Comment by nicolas-raoul** [comment]
No answer from Shabar, so feel free to work on it @jmora13, thanks! Please let us know about your progress every few days :-)

**Comment by bhavanagarlapati** [comment]
Hi @nicolas-raoul Is anyone working on this issue? Can I take it up?

**Comment by samyak1326** [comment]
Hi @nicolas-raoul is anyone working on this issue?? I would like to take this up and work on it.

Thanks

**Comment by nicolas-raoul** [comment]
@samyak1326 Any progress on this? Did you manage to follow the reproduction steps? :-)

**Comment by samyak1326** [comment]
Hi @nicolas-raoul , yes, I was able to follow the steps and reproduce the error, and started working on the issue. It might take some more time to conclude. I will try my best.
